common/static/common/js/spec/discussion/view/new_post_view_spec.js

```diff
@@ -240,6 +240,70 @@
+                    allow_anonymous: true,
+                    category_map: {
+                        subcategories: {
+                            'Week 1': {
+                                subcategories: {},
+                                children: [
+                                    ['Topic-Level Student-Visible Label', 'entry']
+                                ],
+                                entries: {
+                                    'Topic-Level Student-Visible Label': {
+                                        sort_key: null,
+                                        is_cohorted: false,
+                                        id: '2b3a858d0c884eb4b272dbbe3f2ffddd'
+                                    }
+                                }
+                            }
+                        },
+                        children: [
+                            ['First topic', 'entry'],
+                            ['Week 1', 'subcategory']
+                        ],
+                        entries: {
+                            'First topic': {
+                                sort_key: 'First topic',
+                                is_cohorted: false,
+                                id: 'i4x-waqastest-waqastest-course-waqastest'
+                            }
+                        }
+                    }
+                });
+            });
+            cancelPostResetTopic = function(mode, discussion, courseSettings) {
+                var eventSpy, view;
+                view = new NewPostView({
+                    el: $('#fixture-element'),
+                    collection: discussion,
+                    course_settings: courseSettings,
+                    mode: mode
+                });
+                view.render();
+                eventSpy = jasmine.createSpy('eventSpy');
+                view.listenTo(view, 'newPost:cancel', eventSpy);
+                view.$('.post-errors').html("<li class='post-error'>Title can't be empty</li>");
+                view.$("label[for$='post-type-question']").click();
+                view.$('.js-post-title').val('Test Title');
+                view.$('.js-post-body textarea').val('Test body');
+                view.$('.wmd-preview p').html('Test body');
+                view.$('input[name=follow]').prop('checked', false);
+                view.$('input[name=anonymous]').prop('checked', true);
+                view.$('input[name=anonymous_to_peers]').prop('checked', true);
+                view.$("a[data-discussion-id='2b3a858d0c884eb4b272dbbe3f2ffddd']").click();
+                view.$('.cancel').click();
+                expect(eventSpy).toHaveBeenCalled();
+                expect(view.$('.post-topic option:selected').text()).toEqual('First topic');
+            };
+            it('should be the first topic if General is not found', function() {
+                cancelPostResetTopic('tab', this.discussion, this.course_settings);
```

Comment:
Nit: I prefer to keep the assertions in the test, because a written it is hard to see what the test is doing. Helpers are a good idea if they are doing complex setup and/or shared logic, but they shouldn't hide the tests.